src/generators/notebookgenerator/tagstool/index.tsx

```diff
@@ -83,43 +84,63 @@ export class TagsToolComponent extends React.Component<
   }
 
   /*
-  * Selects cells in the document that are tagged with any of the selected tags
-  * in the TOC tags dropdown
+  * Tells the generator to filter the TOC by the selected tags.
   */
-  selectCells = () => {
-    let notebookPanel = this.tracker.currentWidget;
-    if (notebookPanel) {
-      let notebook = notebookPanel.content;
-      let first: boolean = true;
-      for (let i = 0; i < notebookPanel.model.cells.length; i++) {
-        let currentCell = notebook.widgets[i] as Cell;
-        for (let j = 0; j < this.state.selected.length; j++) {
-          if (this.containsTag(this.state.selected[j], currentCell)) {
-            if (first === true) {
-              notebook.deselectAll();
-              notebook.activeCellIndex = i;
-              first = false;
-            } else {
-              notebook.select(notebook.widgets[i] as Cell);
-            }
-          }
-        }
+  filterTags = (selected: string[]) => {
+    this.props.generatorOptionsRef.filtered = selected;
```

Comment:
According to the React docs, a component should never update its own props: https://reactjs.org/docs/components-and-props.html#props-are-read-only

We should consider making the filtered options part of the component state, or passing the `filtered` cells into it from its parent component.